client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs

```diff
@@ -290,6 +290,52 @@ public async Promise CreateService(string name, ServiceType type, List<IBeamoSer
 			}
 		}
 
+		
+		public async Promise UpdateServiceReferences(string serviceName, List<string> assemblyReferencesNames)
+		{
+			LogVerbose($"Starting updating references");
+			//get a list of all references of that service
+			var service = _services.FirstOrDefault(s => s.name == serviceName);
+			if (service == null)
+			{
+				return;
+			}
+
+			LogVerbose($"Reading all references from service: {serviceName}");
+			var results = await _dotnetService.Run($"list {service.CsprojPath} reference");
+
+			//filter that list with only the generated projs
+			var depsPaths = results.Select(s => s.Replace(Environment.NewLine, string.Empty)).Where(line => line.EndsWith(".csproj")).ToList();
+			var correctedPaths = depsPaths.Select(path => path.Replace("\\", "/")).ToList();
+			var existinReferences = correctedPaths.Where(path => path.Contains(CsharpProjectUtil.PROJECT_NAME_PREFIX)).ToList();
+
+			
+			//remove all generated projs references
+			LogVerbose($"Removing all references from service: {serviceName}");
+			foreach (string reference in existinReferences)
+			{
+				var referenceName = Path.GetFileNameWithoutExtension(reference).Replace(CsharpProjectUtil.PROJECT_NAME_PREFIX, String.Empty);
+				var refPathToRemove = CsharpProjectUtil.GenerateCsharpProjectFilename(referenceName);
+				LogVerbose($"Removing reference: {refPathToRemove}");
+				await _dotnetService.Run($"remove {service.CsprojPath} reference {refPathToRemove}");

```

Comment:
is it possible to add these references in parallel? 
Like, instead of `await`'ing each of them in sequence, can we start all invocations, and await the `Promise.Sequence` of them?

Same question for the later addition phase.

Reply:
It is! Thanks for pointing this out